server/src/main/java/org/opensearch/action/admin/indices/shrink/TransportResizeAction.java

```diff
@@ -176,12 +180,22 @@ static CreateIndexClusterStateUpdateRequest prepareCreateIndexRequest(
         targetIndexSettingsBuilder.remove(IndexMetadata.SETTING_HISTORY_UUID);
         final Settings targetIndexSettings = targetIndexSettingsBuilder.build();
         final int numShards;
+
+        // max_shard_size is only supported for shrink
+        ByteSizeValue maxShardSize = resizeRequest.getMaxShardSize();
+        if (resizeRequest.getResizeType() != ResizeType.SHRINK && maxShardSize != null) {
+            throw new IllegalArgumentException("Unsupported parameter [max_shard_size]");
```

Comment:
@reta here we throw an illegalArgumentException for split&clone resize types, because `max_shard_size` is not supported for them.